packages/svelte/tests/signals/test.ts

```diff
@@ -446,4 +447,75 @@ describe('signals', () => {
 			assert.equal(state?.reactions, null);
 		};
 	});
+
+	test('disconnected and reconnected deriveds work as intended', () => {
+		let a: Derived<unknown>;
+		let state = source(0);
+		let log: any[] = [];
+		let b = derived(() => {
+			log.push('b');
+			return $.get(state);
+		});
+		let c = derived(() => {
+			log.push('c');
+			return $.get(b);
+		});
+		let destroy: () => void;
+
+		function connect() {
+			destroy = effect_root(() => {
+				render_effect(() => {
+					a = derived(() => {
+						log.push('a');
+						return $.get(c);
+					});
+					$.get(a);
+				});
+			});
+		}
+
+		return () => {
+			connect();
+			flushSync();
+			assert.deepEqual(log, ['a', 'c', 'b']);
+			log.length = 0;
+			destroy();
+			set(state, 1);
+			$.set_signal_status(c, DIRTY);
```

Comment:
is there a version of this test that doesn't need to use internal APIs? it would help illustrate the case that this change is fixing

Reply:
Fixed